vector/agent/namespaced/daemonset.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: vector
+  labels:
+    app.kubernetes.io/name: vector
+    app.kubernetes.io/instance: vector
+    app.kubernetes.io/component: Agent
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: vector
+      app.kubernetes.io/instance: vector
+      app.kubernetes.io/component: Agent
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: vector
+        app.kubernetes.io/instance: vector
+        app.kubernetes.io/component: Agent
+    spec:
+      serviceAccountName: vector
+      initContainers:
+        - name: render-config
+          image: alpine
+          args:
+            - /bin/sh
+            - -c
+            - apk add --no-cache gettext;
+              envsubst < /etc/config-template/agent.yaml > /etc/config-rendered/agent.yaml
```

Comment:
We are not exporting any envvars?

Reply:
yup, we already discussed that in slack thread. I probably made a syntax mistake when I tried this and fell back to an initContainer and envsubst, but vector should be able to interpolate env vars directly so I will fix